Diabetes/Startup/APHProfileExtender.m

```diff
@@ -119,15 +135,94 @@ - (CGFloat)tableView:(UITableView *)tableView heightForRowAtAdjustedIndexPath:(N
   */
 - (void)navigationController:(UINavigationController *)navigationController didSelectRowAtIndexPath:(NSIndexPath *) __unused indexPath
 {
-    UIStoryboard *sbGlucoseLog = [UIStoryboard storyboardWithName:@"APHOnboarding" bundle:[NSBundle mainBundle]];
-    APHGlucoseLevelsMealTimesViewController *controller = [sbGlucoseLog instantiateViewControllerWithIdentifier:@"APHGlucoseMealTimes"];
+    UIViewController *controller = nil;
+    
+    switch (indexPath.row) {
+        case APHProfileSectionGlucoseLog:
+        {
+            UIStoryboard *sbGlucoseLog = [UIStoryboard storyboardWithName:@"APHOnboarding" bundle:[NSBundle mainBundle]];
+            APHGlucoseLevelsMealTimesViewController *glucoseController = [sbGlucoseLog instantiateViewControllerWithIdentifier:@"APHGlucoseMealTimes"];
+            
+            glucoseController.navigationController.navigationBar.topItem.title = NSLocalizedString(@"Glucose Log", @"");
+            glucoseController.hidesBottomBarWhenPushed = YES;
+            
+            glucoseController.isConfigureMode = YES;
+            
+            controller = glucoseController;
+            
+            [navigationController pushViewController:controller animated:YES];
+        }
+            break;
+            
+        default:
+        {
+            MFMailComposeViewController *feedbackController = [self feedback];
+            
+            self.weakNavController = navigationController;
+            
+            if (feedbackController) {
+                controller = feedbackController;
+                
+                [navigationController presentViewController:controller animated:YES completion:nil];
+            } else {
+                NSString *addEmailAccountMessage = NSLocalizedString(@"You don't have any email accounts set up. Please add an email account in Settings -> Mail, Contacts, Calendars", nil);
+                UIAlertController *alertView = [UIAlertController alertControllerWithTitle:NSLocalizedString(@"No Email Accounts", nil)
+                                                                                   message:addEmailAccountMessage
+                                                                            preferredStyle:UIAlertControllerStyleAlert];
+                
+                UIAlertAction* defaultAction = [UIAlertAction actionWithTitle:@"OK" style:UIAlertActionStyleDefault
```

Comment:
Have you tested this? The last time I played with this, MFMailComposeViewController took care of presenting a system alert for this just fine after its init method returned nil. As I recall, the system dialog has an action that could actually start the email setup process. Open to contradiction if this is out of date.